Source/App/EbAppString.h

```diff
@@ -0,0 +1,172 @@
+*
+* Copyright � 2008, 2009, 2012 Cisco Systems
+* All rights reserved.
+
+* ignore_handler_s.c
+*
+* 2012, Jonathan Toppins <jtoppins@users.sourceforge.net>
+*
+* Copyright � 2012 Cisco Systems
+* All rights reserved.
+*
+* Permission is hereby granted, free of charge, to any person
+* obtaining a copy of this software and associated documentation
+* files (the "Software"), to deal in the Software without
+* restriction, including without limitation the rights to use,
+* copy, modify, merge, publish, distribute, sublicense, and/or
+* sell copies of the Software, and to permit persons to whom the
+* Software is furnished to do so, subject to the following
+* conditions:
+*
+* The above copyright notice and this permission notice shall be
+* included in all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+* OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+* NONINFRINGEMENT.  IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+* HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+* WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+* FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+* OTHER DEALINGS IN THE SOFTWARE.
+*------------------------------------------------------------------
+*/
+#ifndef EbAppString_h
+#define EbAppString_h
+
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif // __cplusplus
+
+#ifndef _RSIZE_T_DEFINED
+typedef size_t rsize_t;
+#define _RSIZE_T_DEFINED
+#endif  /* _RSIZE_T_DEFINED */
+
+#ifndef _ERRNO_T_DEFINED
+#define _ERRNO_T_DEFINED
+typedef int32_t errno_t;
+#endif  /* _ERRNO_T_DEFINED */
+
+#ifndef EOK
+#define EOK             ( 0 )
+#endif
+
+#ifndef ESZEROL
+#define ESZEROL         ( 401 )       /* length is zero              */
+#endif
+
+#ifndef ESLEMIN
+#define ESLEMIN         ( 402 )       /* length is below min         */
+#endif
+
+#ifndef ESLEMAX
+#define ESLEMAX         ( 403 )       /* length exceeds max          */
+#endif
+
+#ifndef ESNULLP
+#define ESNULLP         ( 400 )       /* null ptr                    */
+#endif
+
+#ifndef ESOVRLP
+#define ESOVRLP         ( 404 )       /* overlap undefined           */
+#endif
+
+#ifndef ESEMPTY
+#define ESEMPTY         ( 405 )       /* empty string                */
+#endif
+
+#ifndef ESNOSPC
+#define ESNOSPC         ( 406 )       /* not enough space for s2     */
+#endif
+
+#ifndef ESUNTERM
+#define ESUNTERM        ( 407 )       /* unterminated string         */
+#endif
+
+#ifndef ESNODIFF
+#define ESNODIFF        ( 408 )       /* no difference               */
+#endif
+
+#ifndef ESNOTFND
+#define ESNOTFND        ( 409 )       /* not found                   */
+#endif
+
+#define RSIZE_MAX_STR      ( 4UL << 10 )      /* 4KB */
+#define RCNEGATE(x)  (x)
+
+#ifndef sldebug_printf
+#define sldebug_printf(...)
+#endif
```

Comment:
`sldebug_printf` gets defined to nothing, so `sldebug_printf("IGNORE CONSTRAINT HANDLER: (%u) %s\n", error, (msg) ? msg : "Null message");` gets converted to ` `